noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/note.nr

```diff
@@ -0,0 +1,557 @@
+use crate::{
+    context::PrivateContext,
+    encrypted_logs::encrypt::aes128::derive_aes_symmetric_key_and_iv_from_ecdh_shared_secret_using_sha256,
+    keys::{
+        ecdh_shared_secret::derive_ecdh_shared_secret_using_aztec_address,
+        ephemeral::generate_ephemeral_key_pair,
+    },
+    note::{note_emission::NoteEmission, note_interface::NoteInterface},
+    oracle::{
+        notes::{get_app_tag_as_sender, increment_app_tagging_secret_index_as_sender},
+        random::random,
+    },
+    utils::{bytes::{be_bytes_31_to_fields, get_random_bytes}, point::get_sign_of_point},
+};
+use dep::protocol_types::{
+    abis::note_hash::NoteHash, address::AztecAddress, constants::PRIVATE_LOG_SIZE_IN_FIELDS,
+};
+use std::aes128::aes128_encrypt;
+
+pub(crate) global HEADER_CIPHERTEXT_SIZE_IN_BYTES: u32 = 48; // contract_address (32) + ciphertext_length (2) + 16 bytes pkcs#7 AES padding.
+
+/*
+ * WHY IS THERE LOTS OF CODE DUPLICATION BETWEEN event.nr and note.nr?
+ * It's because there are a few more optimisations that can be done for notes,
+ * and so the stuff that looks like duplicated code currently, won't be
+ * the same for long.
+ * To modularise now feels premature, because we might get that modularisation wrong.
+ * Much better (imo) to have a flattened templates for log assembly, because it
+ * makes it much easier for us all to follow, it serves as a nice example for the
+ * community to follow (if they wish to roll their own log layouts), and it gives
+ * us more time to spot common patterns across all kinds of log layouts.
+ */
+
+/*
+ * LOG CONFIGURATION CHOICES:
+ *
+ * deliver_to: INPUT as recipient: AztecAddress
+ * encrypt_with: aes128 CBC (Cipher Block Chaining)
+ * shared_secret: ephemeral
+ * shared_secret_randomness_extraction_hash: sha256
+ * tag: true
+ * tag_from: INPUT as sender: AztecAddress
+ *
+ * Note-specific:
+ * derive_note_randomness_from_shared_secret: false
+ *
+ */
+
+/*
+ * LOG LAYOUT CHOICE:
+ *
+ * Short explanation:
+ * log = [tag, epk, header_ct=[[contract_address, ct_len], pkcs7_pad], ct=[[pt], pkcs7_pad], some bytes padding, some fields padding]
+ *
+ * Long explanation:
+ * tag: Field
+ * epk: [Field, u8]
+ * header_ct: [[u8; 32], [u8; 2], [u8; 16]]
```

Comment:
It says below that the padding is 14, not 16

Reply:
Ah, 14 is correct.